src/Uno.Extensions.Navigation.Generators/ForceBindingsUpdateGenTool_1.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Uno.Extensions.Generators;
+
+namespace Uno.Extensions.Navigation.Generators;
+
+internal class ForceBindingsUpdateGenTool_1 : ICodeGenTool
+{
+	/// <inheritdoc />
+	public string Version => "1";
+
+	private readonly ForceBindingsUpdateGenerationContext _ctx;
+	private readonly IAssemblySymbol _assembly;
+
+	public ForceBindingsUpdateGenTool_1(ForceBindingsUpdateGenerationContext ctx)
+	{
+		_ctx = ctx;
+		_assembly = ctx.Context.Compilation.Assembly;
+	}
+
+	private bool IsSupported(INamedTypeSymbol? type)
+		=> type is not null
+			&& (_ctx.IsGenerationEnabled(type)
+				?? (
+						type.Is(_ctx.Page) &&
```

Comment:
Is the issue only for `Page`s? What about `UserControl`s?

Reply:
This specific issue is only for Pages as we have to use Frame.Navigate to go to the page. For UserControls we're able to create the control and set the DataContext before adding it to the visual hierarchy (although I will double check this in navigation to verify this behaviour)